src/Bridges/ApplicationDI/RoutingExtension.php

```diff
@@ -55,6 +52,21 @@ public function loadConfiguration()
 	public function beforeCompile()
 	{
 		$builder = $this->getContainerBuilder();
+		$router = $builder->getDefinition($this->prefix('router'));
+
+		if ($this->config['routeClass']) {
+			$class = $this->config['routeClass'];
+		} else {
+			$namespace = Nette\PhpGenerator\Helpers::extractNamespace($router->getFactory()->getEntity());
```

Comment:
In the case that a class `Route` exists in the same NS as RouteListm, yes. But this magic could be removed and used `$this->config['routeClass']` if it is set or `Nette\Application\Routers\Route::class` otherwise. In that case, there should not be any BC.